src/transport.js

```diff
@@ -80,6 +80,9 @@ module.exports = function (swarm) {
       const createListeners = multiaddrs.map((ma) => {
         return (cb) => {
           const listener = transport.createListener(handler)
+
+          listener.once('error', cb)
+
```

Comment:
removing listeners is part of the events API (https://nodejs.org/api/events.html#events_event_removelistener), you essentially just need to passed a named function.

> Shouldn't the consumers know about such errors and be able to react to them as they see appropriate?

There are errors that belong to the 'start listening process' and errors that are 'the socket or the interface where I was listening went away', these need to be handled differently